src/routes/auth/index.ts

```diff
@@ -1,142 +1,288 @@
+import { createRoute, OpenAPIHono } from '@hono/zod-openapi';
 import sgMail from '@sendgrid/mail';
 import { addHours } from 'date-fns';
-import { type Context, Hono } from 'hono';
+import { Hono } from 'hono';
 import { generateEmailHtml } from '../../email-templates/confirm-email.ts';
+import { authorizeVolunteer } from '../../middleware/createMiddleware.ts';
 import type { SessionData } from '../../types/data/session';
 import { hashPassword, validatePassword } from '../../utils/password-hashing.ts';
-import { StatusCodes, type StatusResponse } from '../../utils/responses.ts';
-import { insertProfile } from '../profile/data.ts';
+import { type HeartbeatResponse, HeartbeatResponseSchema, StatusCodes, type StatusResponse, statusResponseFormatter, StatusResponseSchema } from '../../utils/responses.ts';
+import { getProfileById, insertProfile } from '../profile/data.ts';
 import { type CreateProfileRequestBody, CreateProfileSchema } from '../profile/validation.ts';
 import { activateProfile, checkEmail, getLoginInfo } from './data.ts';
-import { type SignInData, SignInSchema } from './validate.ts';
-
+import { ActivateSchema, type SignInData, SignInSchema, SignUpSchema } from './validate.ts';
 export const authRoutes = new Hono();
 
-authRoutes.post('/sign-up', async (context: Context<EnvironmentBindings>) => {
-	let parsedBody: CreateProfileRequestBody;
+// Public Routes (Post, Get)
+export const publicAuthRoutes = new OpenAPIHono<EnvironmentBindings>({
+	defaultHook: statusResponseFormatter
+});
 
-	try {
-		const body = await context.req.json();
-		parsedBody = CreateProfileSchema.parse(body);
-	} catch (error) {
-		if (error instanceof SyntaxError) {
-			return context.json<StatusResponse>({ message: `Invalid JSON format: ${error.message}` }, StatusCodes.BAD_REQUEST);
+publicAuthRoutes.openapi(
+	createRoute({
+		method: 'post',
+		path: '/',
+		operationId: 'signUp',
+		description: 'Sign up to community hub account',
+		tags: ['Sign-up'],
+		request: {
+			body: { content: { 'application/json': { schema: SignUpSchema } }, required: true }
```

Comment:
suggestion: I don't think the `/auth/sign-up` route should only accept email & password. It also needs to take in `name` and `description` for example to create a profile. Instead of using `SignUpSchema` maybe you could use `CreateProfileSchema` instead?

Reply:
It should have name(is missing), email and password.  Create Profile Schema is a different flow.

Reply:
@abdulqshabbir @madcampos

Reply:
Oh oops maybe I may be wrong here sorry. I can double check in the morning. Maybe I was thinking create profile route and not sign up route.

Reply:
I double-checked the dev branch and for the `/auth/sign-up` route it is using `CreateProfileSchema` to parse the request body. It seems that the profile is also created in the same route (whether we want that or not I'm not sure I'm just listing the existing functionality)

Reply:
I might be lost in semantics here. 

![image](https://github.com/user-attachments/assets/ed5ef4a0-b1a3-44fb-ae46-c30acfcfa99c)

Reply:
The sign-up route has only email and password. It should create a new profile, if one does not exist and silently fail otherwise. (It is an attack vector to try and create a new profile with an existing email just to check if it exists or not).

**Then** there is the complete profile step where more information will be provided.